src/lib.rs

```diff
@@ -358,5 +359,13 @@ bitflags::bitflags! {
         /// - `+++` line at start
         /// - `+++` line at end
         const ENABLE_PLUSES_DELIMITED_METADATA_BLOCKS = 1 << 8;
+        /// GitHub-compatible footnote syntax. Mutually-exclusive with `ENABLE_FOOTNOTES`.
+        const ENABLE_GFM_FOOTNOTES = 1 << 9;
```

Comment:
I think that nothing prevents here from enabling both options, which is incorrect. This is a special case because there are two different options for the same thing with small differences.

Pulldown-cmark adheres to CommonMark and GFM extensions, so I think the best solution here is maintaining the `ENABLE_FOOTNOTES` flag enabling the new behaviour and a new `FOOTNOTES_OLD_STYLE` or something similar to enable the old behaviour not fully compatible with GFM footnotes.

This supposes a breaking change, but 0.10 release contains some of them, so I would prioritize GFM adhesion against not breaking changes here.

Reply:
What do you think, @raphlinus?